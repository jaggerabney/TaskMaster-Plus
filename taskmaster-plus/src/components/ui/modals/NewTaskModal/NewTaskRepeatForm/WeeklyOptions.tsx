import Button from "@/components/ui/Button";

import { WeeklyOptionsType, daysOfTheWeek } from "@/utils/Util";

interface WeeklyOptionsPropsType {
  data: WeeklyOptionsType;
  onChange: (day: string) => void;
}

const WeeklyOptions: React.FC<WeeklyOptionsPropsType> = ({
  data,
  onChange
}) => {
  const { byDay } = data;
  const weekdays = daysOfTheWeek.map((day) => day.slice(0, 3));

  return (
    <ul className="flex flex-row justify-between gap-2 ml-8 mr-8">
      {weekdays.map((day) => (
        <Button
          key={day}
          text={day}
          type="button"
          textColor={byDay.includes(day) ? "text-white" : "text-redNCS"}
          borderColor="border-redNCS"
          bgColor={byDay.includes(day) ? "bg-redNCS" : "bg-white"}
          onClick={() => onChange(day)}
        />
      ))}
    </ul>
  );
};

export default WeeklyOptions;
