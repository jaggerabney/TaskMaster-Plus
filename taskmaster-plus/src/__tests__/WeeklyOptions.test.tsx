import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import WeeklyOptions from "@/components/ui/modals/NewTaskModal/NewTaskRepeatForm/WeeklyOptions";

describe("Weekly options component", () => {
  const mockOnSubmit = jest.fn();
  const mockWeeklyOptions = (
    <WeeklyOptions data={{ byDay: [] }} onChange={mockOnSubmit} />
  );

  it("should render correctly", () => {
    const weeklyOptions = render(mockWeeklyOptions);

    expect(weeklyOptions).toMatchSnapshot();
  });

  it("allows user to change values", async () => {
    render(mockWeeklyOptions);
    const thuButton = screen.getByText("Thu");
    await userEvent.click(thuButton);

    const onChangeData: string = mockOnSubmit.mock.calls[0][0];
    expect(onChangeData).toEqual("Thu");
  });

  it("should have no days selected by default", () => {
    render(mockWeeklyOptions);
    const weeklyOptions = document.getElementById(
      "weekly-options"
    ) as HTMLUListElement;

    for (const option of weeklyOptions.children) {
      expect(option).toHaveClass("text-redNCS");
    }
  });
});
