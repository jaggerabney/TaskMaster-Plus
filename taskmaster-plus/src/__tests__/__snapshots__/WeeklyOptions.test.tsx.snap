// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Weekly options component should render correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <ul
        class="flex flex-row justify-between gap-2 ml-8 mr-8"
        id="weekly-options"
      >
        <button
          class="px-4 py-2 w-full text-redNCS border-redNCS bg-white whitespace-nowrap select-none font-bold border-2 rounded hover:scale-[1.03] duration-100"
          type="button"
        >
          Sun
        </button>
        <button
          class="px-4 py-2 w-full text-redNCS border-redNCS bg-white whitespace-nowrap select-none font-bold border-2 rounded hover:scale-[1.03] duration-100"
          type="button"
        >
          Mon
        </button>
        <button
          class="px-4 py-2 w-full text-redNCS border-redNCS bg-white whitespace-nowrap select-none font-bold border-2 rounded hover:scale-[1.03] duration-100"
          type="button"
        >
          Tue
        </button>
        <button
          class="px-4 py-2 w-full text-redNCS border-redNCS bg-white whitespace-nowrap select-none font-bold border-2 rounded hover:scale-[1.03] duration-100"
          type="button"
        >
          Wed
        </button>
        <button
          class="px-4 py-2 w-full text-redNCS border-redNCS bg-white whitespace-nowrap select-none font-bold border-2 rounded hover:scale-[1.03] duration-100"
          type="button"
        >
          Thu
        </button>
        <button
          class="px-4 py-2 w-full text-redNCS border-redNCS bg-white whitespace-nowrap select-none font-bold border-2 rounded hover:scale-[1.03] duration-100"
          type="button"
        >
          Fri
        </button>
        <button
          class="px-4 py-2 w-full text-redNCS border-redNCS bg-white whitespace-nowrap select-none font-bold border-2 rounded hover:scale-[1.03] duration-100"
          type="button"
        >
          Sat
        </button>
      </ul>
    </div>
  </body>,
  "container": <div>
    <ul
      class="flex flex-row justify-between gap-2 ml-8 mr-8"
      id="weekly-options"
    >
      <button
        class="px-4 py-2 w-full text-redNCS border-redNCS bg-white whitespace-nowrap select-none font-bold border-2 rounded hover:scale-[1.03] duration-100"
        type="button"
      >
        Sun
      </button>
      <button
        class="px-4 py-2 w-full text-redNCS border-redNCS bg-white whitespace-nowrap select-none font-bold border-2 rounded hover:scale-[1.03] duration-100"
        type="button"
      >
        Mon
      </button>
      <button
        class="px-4 py-2 w-full text-redNCS border-redNCS bg-white whitespace-nowrap select-none font-bold border-2 rounded hover:scale-[1.03] duration-100"
        type="button"
      >
        Tue
      </button>
      <button
        class="px-4 py-2 w-full text-redNCS border-redNCS bg-white whitespace-nowrap select-none font-bold border-2 rounded hover:scale-[1.03] duration-100"
        type="button"
      >
        Wed
      </button>
      <button
        class="px-4 py-2 w-full text-redNCS border-redNCS bg-white whitespace-nowrap select-none font-bold border-2 rounded hover:scale-[1.03] duration-100"
        type="button"
      >
        Thu
      </button>
      <button
        class="px-4 py-2 w-full text-redNCS border-redNCS bg-white whitespace-nowrap select-none font-bold border-2 rounded hover:scale-[1.03] duration-100"
        type="button"
      >
        Fri
      </button>
      <button
        class="px-4 py-2 w-full text-redNCS border-redNCS bg-white whitespace-nowrap select-none font-bold border-2 rounded hover:scale-[1.03] duration-100"
        type="button"
      >
        Sat
      </button>
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
